src/main/java/com/github/manolo8/darkbot/core/manager/GuiManager.java

```diff
@@ -215,14 +215,13 @@ public boolean canTickModule() {
 
 
         HeroManager hero = main.hero;
-        if (main.config.MISCELLANEOUS.REFRESH_AFTER_REVIVE
-                && this.needRefresh
-                && System.currentTimeMillis() - lastRepair > 5_000
-                && hero.health.hp >= 1_000 ){
-            System.out.println("Triggering refresh: refreshing after death");
-            API.handleRefresh();
+        if (this.needRefresh && System.currentTimeMillis() - lastRepair > 5_000){
+            if(main.config.MISCELLANEOUS.REFRESH_AFTER_REVIVE ){
+                System.out.println("Triggering refresh: refreshing after death");
+                API.handleRefresh();
+                return false;
```

Comment:
```suggestion
```
you still need to set the need refresh to false, and it will resturn false on the line below anyways